edgelet/edgelet-http-workload/src/server/cert/mod.rs

```diff
@@ -62,34 +67,178 @@ fn compute_validity(expiration: &str, max_duration_sec: i64, context: ErrorKind)
     Ok(cmp::min(secs, max_duration_sec))
 }
 
-fn refresh_cert<T: CreateCertificate>(
-    hsm: &T,
+fn refresh_cert(
+    key_client: Arc<Mutex<aziot_key_client::Client>>,
+    cert_client: Arc<Mutex<CertificateClient>>,
     alias: String,
     props: &CertificateProperties,
     context: ErrorKind,
-) -> Result<Response<Body>> {
-    if let Err(err) = hsm.destroy_certificate(alias) {
-        return Err(Error::from(err.context(context)));
-    };
+) -> Box<dyn Future<Item = Response<Body>, Error = HttpError> + Send> {
+    let response = generate_key_and_csr(props, context.clone())
+        .map_err(|_| Error::from(ErrorKind::CertOperation(CertOperation::CreateIdentityCert)))
+        .and_then(|(privkey, csr)| {
+            let workload_ca_key_pair_handle = key_client.lock().expect("key client lock error").create_key_pair_if_not_exists(IOTEDGED_CA_ALIAS, Some("ec-p256:rsa-4096:*"))
```

Comment:
The convention iotedged follows is that the workload CA type (EC vs RSA) follows the device CA type, and module certs types follow the workload CA type.

Reply:
This should be load_key because it is preloaded

Reply:
Workload CA is not preloaded, let alone its key pair.

Reply:
As discussion on this front, I will follow up on whether it makes sense to just keep device CA, since the workload CA has nearly all the same properties as its parent in iotedge 1.0.10.